packages/plugin-core/src/test/suite-integ/SyncCommand.test.ts

```diff
@@ -390,6 +396,71 @@ suite("workspace sync command", function () {
       });
     });
   });
+
+  describe("WHEN there are tracked, uncommitted changes", () => {
+    test("THEN Dendron stashes and restores the changes", (done) => {
```

Comment:
also add tests for when pull fails. we want to be more diligent with tests - they don't need to be exhaustive but we should test at least the happy case and 2-3 edge cases

Reply:
What do you mean by "pull fails"? Something where we can't pull at all (like a permissions issue), or the rebase hitting a conflict? I'll add tests for these, let me know if you had anything else in mind.

Reply:
rebase hitting a conflict is the main case i'm thinking of